External/FEXCore/Source/Interface/Core/JIT/Arm64/JIT.cpp

```diff
@@ -791,6 +791,10 @@ CPUBackend::CompiledCode Arm64JITCore::CompileCode(uint64_t Entry,
   adr(TMP1, &JITCodeHeaderLabel);
   str(TMP1, STATE, offsetof(FEXCore::Core::CPUState, InlineJITBlockHeader));
 
+  // Trigger a fault if there are any pending interrupts
+  strb(ARMEmitter::XReg::zr, STATE, offsetof(FEXCore::Core::InternalThreadState, InterruptFaultPage) -
```

Comment:
I believe you'll want to use 64-bit str here. Since the imm offset has significant offset limitations since it gets scaled by the size of the access. Theoretically this strb won't even work here (and should assert in assertions builds).
- strb imm access range is 0-4095
- str (64-bit) imm access range is 0-32760 (members must be aligned to 64-bit natural alignment).

64-bit will let us keep scaling the state past 4096 once the base CPUState eventually gets larger than 1 page as well.

Reply:
I understand the range limits, but why would the emitter assert here given it is in range (which I do static assert for)?

Reply:
Oh, you're right. I forgot that the `CpuStateFrame` isn't at the base of `InternalThreadState` anymore. So yes, you do get away with it for now.
Once we add more stuff to the CpuStateFrame we can easily bump it to an strh or whatever.